ssha/config.py

```diff
@@ -125,6 +129,35 @@ def get(key, default=None):
     return render(value)
 
 
+def generate_key():
+    """
+    Generates and adds a key to the SSH agent.
+
+    """
+    key = rsa.generate_private_key(
+        backend=crypto_default_backend(),
+        public_exponent=65537,
+        key_size=4096,
+    )
+    private_key = key.private_bytes(
+        crypto_serialization.Encoding.PEM,
+        crypto_serialization.PrivateFormat.PKCS8,
+        crypto_serialization.NoEncryption(),
+    )
+    public_key = key.public_key().public_bytes(
+        crypto_serialization.Encoding.OpenSSH,
+        crypto_serialization.PublicFormat.OpenSSH,
+    )
+
+    with tempfile.NamedTemporaryFile() as temp_key:
```

Comment:
I've done `tempfile.NamedTemporaryFile(suffix='-ssha-known-hosts')` elsewhere. Can you please use a similar suffix here, maybe `-ssha-temporary-key`? It just makes the file names more obvious if it crashes or you're debugging things.